src/unraphael/dash/pages/5_cluster.py

```diff
@@ -0,0 +1,298 @@
+from __future__ import annotations
+import numpy as np
+import streamlit as st
+#from skimage import color
+from unraphael.dash.image_clustering import (align_images_to_mean, equalize_images,
+                                             cluster_images, compute_metrics,
+                                             clustimage_clustering, plot_clusters,
+                                             plot_dendrogram, matrix_of_similarities)
+from styling import set_custom_css
+from widgets import load_images_widget, show_heatmaps_widget
+#import matplotlib.pyplot as plt
+#import seaborn as sns
+#import io
+
+def show_images_widget(
+    images: dict[str, np.ndarray],
+    *,
+    n_cols: int = 4,
+    key: str = 'show_images',
+    message: str = 'Select image',
+) -> None | str:
+    """Widget to show images with given number of columns."""
+    col1, col2 = st.columns(2)
+    
+    # Ensure each number_input has a unique key
+    n_cols_key = f'{key}_cols'
+    n_cols = col1.number_input(
+        'Number of columns for display', value=4, min_value=1, step=1, key=n_cols_key
+    )    
+
+    cols = st.columns(n_cols)
+
+    for i, (name, im) in enumerate(images.items()):
+        if i % n_cols == 0:
+            cols = st.columns(n_cols)
+        col = cols[i % n_cols]
+
+        col.image(im, use_column_width=True, caption=name,clamp=True)    
```

Comment:
Import from `widgets.py`

```suggestion
```

Reply:
Yes, good idea, except that here we do not have to select an image per se. We could of course build that option in in the code for show_images_widget in widgets.py